Build intermittently fails with "explicit type is missing"

The error on failure:

```
/private/home/ezyang/ATen/lib/THC/generic/THCTensorMathPairwise.cu(47): error: explicit type is missing ("int" assumed)

/private/home/ezyang/ATen/lib/THC/generic/THCTensorMathPairwise.cu(58): error: explicit type is missing ("int" assumed)

2 errors detected in the compilation of "/tmp/tmpxft_000076f6_00000000-7_THCTensorMathPairwise.cpp1.ii".
CMake Error at THC_generated_THCTensorMathPairwise.cu.o.cmake:267 (message):
  Error generating file
  /private/home/ezyang/ATen/build/lib/THC/CMakeFiles/THC.dir//./THC_generated_THCTensorMathPairwise.cu.o
```

It seems to go away if you run make again. It seems to happen fairly reproduceably on a fresh build.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,13 +17,7 @@ if("${isSystemDir}" STREQUAL "-1")
 endif()
 
 # C++11
-if(CMAKE_VERSION VERSION_LESS "3.1")
-  set(CMAKE_CXX_FLAGS "--std=c++11 ${CMAKE_CXX_FLAGS}")
-else()
-  set(CMAKE_CXX_STANDARD 11)
-endif()
-
-set(CUDA_NVCC_FLAGS "--std=c++11")
+set(CMAKE_CXX_FLAGS "--std=c++11 ${CMAKE_CXX_FLAGS}")
 
 # TH stuff
 set(Torch_FOUND 1)

